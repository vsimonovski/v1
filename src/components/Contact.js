/* eslint-disable */

import React from 'react';
import './Contact.scss';

export default function Contact() {
  return (
    <section className="contact">
      <div className="item">
        <span className="emoji">👨‍💻</span>
        <a
          href="https://www.linkedin.com/in/vsimonovski/"
          className="value"
          data-wenk="👈  Where I work"
          data-wenk-pos="right"
        >
          @Hooloovoo
        </a>
      </div>
      <div className="item">
        <span className="emoji">💻</span>
        <a
          href="https://github.com/vsimonovski"
          className="value"
          data-wenk="👈  Check out my GitHub"
          data-wenk-pos="right"
        >
          @vsimonovski
        </a>
      </div>
      <div className="item">
        <span className="emoji">🗒️</span>
        <a
          href="http://blog.vsimonovski.me/"
          className="value"
          data-wenk="👈  Sometimes I like to write about code"
          data-wenk-pos="right"
        >
          @blog.vsimonovski
        </a>
      </div>
      <div className="item">
        <span className="emoji">💌</span>
        <a href="mailto:vsimonovski@outlook.com" className="value">
          Contact me?
        </a>
      </div>
    </section>
  );
}
