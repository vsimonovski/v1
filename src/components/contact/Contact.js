import React from 'react';
import {
  ContactItem,
  ContactLink,
  ContactSection,
  Emoji
} from './Contact.style';

function Contact() {
  return (
    <ContactSection>
      <ContactItem>
        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
        <Emoji>👨‍💻</Emoji>
        <ContactLink
          href="https://www.linkedin.com/in/vsimonovski/"
          data-wenk="👈  I work here"
          data-wenk-pos="right"
        >
          @Hooloovoo
        </ContactLink>
      </ContactItem>
      <ContactItem>
        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
        <Emoji>💻</Emoji>
        <ContactLink
          href="https://github.com/vsimonovski"
          data-wenk="👈  Check out my GitHub"
          data-wenk-pos="right"
        >
          @vsimonovski
        </ContactLink>
      </ContactItem>
      <ContactItem>
        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
        <Emoji>🗒️</Emoji>
        <ContactLink
          href="http://blog.vsimonovski.me/"
          data-wenk="👈  Sometimes I like to write about code"
          data-wenk-pos="right"
        >
          @blog.vsimonovski
        </ContactLink>
      </ContactItem>
      <ContactItem>
        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
        <Emoji>💌</Emoji>
        <ContactLink href="mailto:vsimonovski@outlook.com">
          Contact me?
        </ContactLink>
      </ContactItem>
    </ContactSection>
  );
}

export default Contact;
