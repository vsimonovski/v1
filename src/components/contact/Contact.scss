$color__gold: #ffca22;

.contact {
  padding-bottom: 15px;
  padding-top: 15px;
  .item {
    margin-bottom: 18px;
    display: flex;
    align-items: center;
    &:last-child {
      margin-bottom: 0;
    }
  }

  .emoji {
    font-size: 2.2rem;
    padding: 0;
  }

  .value {
    cursor: pointer;
    color: $color__gold;
    font-size: 1.6rem;
    margin-left: 18px;
    border-bottom: 0px solid transparent;
    transition: border 0.2s ease-in;
    text-decoration: none;

    &:hover {
      border-bottom: 1px solid $color__gold;
    }
  }
}
