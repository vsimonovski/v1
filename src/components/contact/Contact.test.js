import React from 'react';
import { render } from '@testing-library/react';
import Contact from './Contact';

it('should match snapshot', () => {
  const tree = render(<Contact />);
  expect(tree).toMatchSnapshot();
});

it('should display contact items', () => {
  const { getByText } = render(<Contact />);

  const workplace = getByText(/@Hooloovoo/i);
  const githubHandle = getByText(/@vsimonovski/i);
  const blogHandle = getByText(/@blog.vsimonovski/i);
  const contactHandle = getByText(/Contact me?/i);

  expect(workplace).toBeInTheDocument();
  expect(githubHandle).toBeInTheDocument();
  expect(blogHandle).toBeInTheDocument();
  expect(contactHandle).toBeInTheDocument();
});
