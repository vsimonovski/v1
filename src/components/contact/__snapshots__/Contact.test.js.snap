// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-bdnylx igwDSu"
      >
        <div
          class="sc-gtssRu bFciFR"
        >
          <span
            class="sc-dlnjPT fSbNAx"
          >
            üë®‚Äçüíª
          </span>
          <a
            class="sc-hKFyIo kQbcBB"
            data-wenk="üëà  I work here"
            data-wenk-pos="right"
            href="https://www.linkedin.com/in/vsimonovski/"
          >
            @Hooloovoo
          </a>
        </div>
        <div
          class="sc-gtssRu bFciFR"
        >
          <span
            class="sc-dlnjPT fSbNAx"
          >
            üíª
          </span>
          <a
            class="sc-hKFyIo kQbcBB"
            data-wenk="üëà  Check out my GitHub"
            data-wenk-pos="right"
            href="https://github.com/vsimonovski"
          >
            @vsimonovski
          </a>
        </div>
        <div
          class="sc-gtssRu bFciFR"
        >
          <span
            class="sc-dlnjPT fSbNAx"
          >
            üóíÔ∏è
          </span>
          <a
            class="sc-hKFyIo kQbcBB"
            data-wenk="üëà  Sometimes I like to write about code"
            data-wenk-pos="right"
            href="http://blog.vsimonovski.me/"
          >
            @blog.vsimonovski
          </a>
        </div>
        <div
          class="sc-gtssRu bFciFR"
        >
          <span
            class="sc-dlnjPT fSbNAx"
          >
            üíå
          </span>
          <a
            class="sc-hKFyIo kQbcBB"
            href="mailto:vsimonovski@outlook.com"
          >
            Contact me?
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-bdnylx igwDSu"
    >
      <div
        class="sc-gtssRu bFciFR"
      >
        <span
          class="sc-dlnjPT fSbNAx"
        >
          üë®‚Äçüíª
        </span>
        <a
          class="sc-hKFyIo kQbcBB"
          data-wenk="üëà  I work here"
          data-wenk-pos="right"
          href="https://www.linkedin.com/in/vsimonovski/"
        >
          @Hooloovoo
        </a>
      </div>
      <div
        class="sc-gtssRu bFciFR"
      >
        <span
          class="sc-dlnjPT fSbNAx"
        >
          üíª
        </span>
        <a
          class="sc-hKFyIo kQbcBB"
          data-wenk="üëà  Check out my GitHub"
          data-wenk-pos="right"
          href="https://github.com/vsimonovski"
        >
          @vsimonovski
        </a>
      </div>
      <div
        class="sc-gtssRu bFciFR"
      >
        <span
          class="sc-dlnjPT fSbNAx"
        >
          üóíÔ∏è
        </span>
        <a
          class="sc-hKFyIo kQbcBB"
          data-wenk="üëà  Sometimes I like to write about code"
          data-wenk-pos="right"
          href="http://blog.vsimonovski.me/"
        >
          @blog.vsimonovski
        </a>
      </div>
      <div
        class="sc-gtssRu bFciFR"
      >
        <span
          class="sc-dlnjPT fSbNAx"
        >
          üíå
        </span>
        <a
          class="sc-hKFyIo kQbcBB"
          href="mailto:vsimonovski@outlook.com"
        >
          Contact me?
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
